### DISCUSSION TITLE
how to suppress specific warnings

### DISCUSSION THREAD

**Discussion by gkamendje** [The-OpenROAD-Project/OpenROAD#3445] [Q&A] [answered]
I would like to temporarily suppress some warnings like `[WARNING STA-0122]`. Is there a way to do that? Is there a place to look for the full meaning of warnings and error messages? Some of them are cryptic.

**Comment by maliberty** [comment]
There is no suppression mechanism.  There is no catalog of messages though you can always look at the source code to get more ideas.  Is there a particular message that you are finding cryptic?

**Reply by gkamendje** [reply]
I am getting tons of messages like the following
`[WARNING STA-0122] ./platforms/myplatform/lib/1.2v/mylib_ccs.lib line 7445, table template my_table1x1x5 not found.`
These are libraries issues and there is nothing I can do about it at this point. I was wondering if I could at least suppress them.

**Reply by gkamendje** [reply]
BTW I do see `[WARNING STA-0122] message limit reached, this message will no longer print` which suggests that there is a limit set somewhere for this message. Any hint on how to change this limit?

**Reply by rovinski** [reply]
@maliberty can't `suppress_message STA 0122` be used here?

**Reply by gkamendje** [reply]
@rovinski yeah I was looking exactly for something like that. I would like to give it a try. Should I put in the design config.mk file?

**Reply by rovinski** [reply, marked as answer]
Libraries are loaded in [`read_liberty.tcl`](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/42e5a0605f18e2e29b359da8086ddb3172ed4794/flow/scripts/read_liberty.tcl) so you can try putting it there.

**Reply by gkamendje** [reply]
BTW, the warning messages I was getting were related to ccs.lib files. I read somewhere that ORFS only supports NLDM for now. Wouldn't it be a good idea to issue an error message when a non NLDM library format is loaded?

**Reply by QuantamHD** [reply]
That's not a bad idea especially if people unintentionally believe its using CCS.

**Reply by gkamendje** [reply]
@rovinski thanks for the hint. Assuming I want to disable some messages on a per design basis. Any hint on how to proceed?

**Reply by rovinski** [reply]
That's a good question, I don't think the flow is really set up for that as of today. A quick hack might be to add
```
if [$::env(DESIGN_NAME) == "your_design_name"] {
  suppress_message STA 0122
}
```
to `read_liberty.tcl`, but that's not a very scalable solution.

Something cleaner would require adding plugin script hooks, which is a big change.

**Reply by gkamendje** [reply]
@rovinski thanks a lot. That's already a very good starting point. Doing it this way certainly prevents global suppression of messages which might be dangerous.

**Reply by maliberty** [reply]
You could put it in your ~/.openroad file if you always want it to be suppressed.

**Comment by maliberty** [comment]
I forgot we had that.  I wrote it with unit tests in mind but I guess it works here too.

**Reply by vijayank88** [reply]
In ORFS we've implemented this: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/42e5a0605f18e2e29b359da8086ddb3172ed4794/flow/scripts/read_liberty.tcl#L4
